### PR TITLE
github: log the response from GH when error occurs

### PR DESCRIPTION
**Pull request by brig**
example:
```
java.lang.RuntimeException: Cannot merge the pull request: Method Not Allowed (405)
 response: {"message":"Pull Request is not mergeable","documentation_url":"https://docs.github.com/rest/pulls/pulls#merge-a-pull-request","status":"405"}
```

### CODE REVIEW

**Review by benbroadaway — DISMISSED** [review]

File: tasks/git/src/main/java/com/walmartlabs/concord/plugins/git/GitHubTask.java [hunk 1]
```diff
@@ -1022,6 +1023,59 @@ private static Map<String, Object> makeResult(Object data) {
...
+                    }
+
+                    if (error != null) {
+                        return new RequestException(error, code);
+                    }
+                } else {
+                    try {
+                        responseBody = new String(response.readAllBytes());
+
+                        response.close();
```
Comment (ibodrov) [inline tasks/git/src/main/java/com/walmartlabs/concord/plugins/git/GitHubTask.java:1057]: Move to `finally`?

**Review by ibodrov — APPROVED** [review]
Minor comment, otherwise LGTM.

File: tasks/git/src/main/java/com/walmartlabs/concord/plugins/git/GitHubTask.java (hunk 1 above)
Comment (benbroadaway) [reply tasks/git/src/main/java/com/walmartlabs/concord/plugins/git/GitHubTask.java]: Gave it the ol' try-with-resources treatment.

Comment (brig) [reply tasks/git/src/main/java/com/walmartlabs/concord/plugins/git/GitHubTask.java]: it was copy-paste from original jgit(?) code :)

**Review by amithkb — APPROVED** [review]

**Pull request merged**